=== modified file 'src/diagnostics/Makefile.am'
--- activity-log-manager-0.9.4.orig/src/diagnostics/Makefile.am	2012-03-01 21:25:48 +0000
+++ activity-log-manager-0.9.4/src/diagnostics/Makefile.am	2012-04-11 12:55:18 +0000
@@ -20,7 +20,7 @@
 	$(POLKIT_LIBS) \
 	$(GIO_UNIX_LIBS)
 
-policykit_in_files = \ 
+policykit_in_files = \
 	com.ubuntu.whoopsiepreferences.policy.in
 
 EXTRA_DIST = \

=== modified file 'src/diagnostics/whoopsie-preferences.c'
--- activity-log-manager-0.9.4.orig/src/diagnostics/whoopsie-preferences.c	2012-02-20 20:27:33 +0000
+++ activity-log-manager-0.9.4/src/diagnostics/whoopsie-preferences.c	2012-04-11 12:51:05 +0000
@@ -122,6 +122,16 @@
     whoopsie_preferences_load_configuration ();
 }
 
+void
+whoopsie_preferences_notify_upstart (gboolean run_whoopsie)
+{
+    GError* error = NULL;
+    char* command[] = { "/sbin/initctl", run_whoopsie ? "start" : "stop", "whoopsie", NULL };
+    if (!g_spawn_sync (NULL, command, NULL, 0, NULL, NULL, NULL, NULL, NULL, &error)) {
+        g_print ("Could not run initctl: %s\n", error->message);
+    }
+}
+
 gboolean
 whoopsie_preferences_changed (WhoopsiePreferences* object, GParamSpec* pspec,
                               gpointer user_data)
@@ -160,6 +170,7 @@
             return FALSE;
         }
     }
+    whoopsie_preferences_notify_upstart (new_value);
 }

 static gboolean

