Index: qt4-x11-4.8.5+git192-g085f851+dfsg/src/declarative/qml/qdeclarativeengine.cpp
===================================================================
--- qt4-x11-4.8.5+git192-g085f851+dfsg.orig/src/declarative/qml/qdeclarativeengine.cpp	2013-09-13 17:31:52.000000000 +0200
+++ qt4-x11-4.8.5+git192-g085f851+dfsg/src/declarative/qml/qdeclarativeengine.cpp	2014-02-25 14:36:03.000000000 +0100
@@ -2555,4 +2555,10 @@
     return true;
 }
 
+QDeclarativeAccessibilityUpdater *QDeclarativeEnginePrivate::getAccessibilityUpdater(QDeclarativeEngine *e)
+{
+    static QDeclarativeAccessibilityUpdater accessibilityUpdater;
+    return &accessibilityUpdater;
+}
+
 QT_END_NAMESPACE
Index: qt4-x11-4.8.5+git192-g085f851+dfsg/src/declarative/qml/qdeclarativeengine_p.h
===================================================================
--- qt4-x11-4.8.5+git192-g085f851+dfsg.orig/src/declarative/qml/qdeclarativeengine_p.h	2014-02-25 14:36:03.000000000 +0100
+++ qt4-x11-4.8.5+git192-g085f851+dfsg/src/declarative/qml/qdeclarativeengine_p.h	2014-02-25 14:39:20.545913220 +0100
@@ -238,8 +238,6 @@
 
     mutable QMutex mutex;
 
-    QDeclarativeAccessibilityUpdater accessibilityUpdater;
-
     QDeclarativeTypeLoader typeLoader;
     QDeclarativeImportDatabase importDatabase;
 
@@ -314,7 +312,7 @@
     static QScriptValue formatTime(QScriptContext*, QScriptEngine*);
     static QScriptValue formatDateTime(QScriptContext*, QScriptEngine*);
 #endif
-    static QDeclarativeAccessibilityUpdater *getAccessibilityUpdater(QDeclarativeEngine *e) { return &e->d_func()->accessibilityUpdater; }
+    static QDeclarativeAccessibilityUpdater *getAccessibilityUpdater(QDeclarativeEngine *e);
     static QScriptEngine *getScriptEngine(QDeclarativeEngine *e) { if (!e) return 0; return &e->d_func()->scriptEngine; }
     static QDeclarativeEngine *getEngine(QScriptEngine *e) { return static_cast<QDeclarativeScriptEngine*>(e)->p->q_func(); }
     static QDeclarativeEnginePrivate *get(QDeclarativeEngine *e) { return e->d_func(); }
